### PR TITLE
Add the "macos" alias for the OSX constructor of the OS datatype

### PR DESCRIPTION
**Pull request by Kleidukos**
---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [x] The documentation has been updated, if necessary.

## Before
Screenshot 2023-02-21 at 10-32-56 Distribution System

## After
Screenshot 2023-02-21 at 10-33-06 Distribution System

### CODE REVIEW

File: Cabal-syntax/src/Distribution/System.hs [hunk 1]
```diff
@@ -88,8 +88,9 @@ data ClassificationStrictness = Permissive | Compat | Strict
 --  ,HaLVM ,Hurd ,IOS, Android, Ghcjs, Wasi
```
Comment (ffaf1) [inline Cabal-syntax/src/Distribution/System.hs:88]: Commas are all over the place (e.g. “,IOS, Android”). You fixed the aliases part, fixing this too would be ace.

Comment (ffaf1) [inline Cabal-syntax/src/Distribution/System.hs:88]: Commas are all over the place (e.g. “,IOS, Android”). You fixed the aliases part, fixing this too would be ace.

**Review by ffaf1 — APPROVED** [review]

File: Cabal-syntax/src/Distribution/System.hs (hunk 1 above)
Comment (Kleidukos) [reply Cabal-syntax/src/Distribution/System.hs:88]: :+1:

File: Cabal-syntax/src/Distribution/System.hs [hunk 2]
```diff
@@ -82,14 +82,14 @@ data ClassificationStrictness = Permissive | Compat | Strict
...
 
--- | These are the known OS names: Linux, Windows, OSX
---  ,FreeBSD, OpenBSD, NetBSD, DragonFly
---  ,Solaris, AIX, HPUX, IRIX
---  ,HaLVM ,Hurd ,IOS, Android, Ghcjs, Wasi
+-- | These are the known OS names: Linux, Windows, OSX, FreeBSD, OpenBSD,
+--   NetBSD, DragonFly, Solaris, AIX, HPUX, IRIX, HaLVM, Hurd, IOS, Android,
+--   Ghcjs, Wasi
 --
 -- The following aliases can also be used:,
```
Comment (robx) [inline Cabal-syntax/src/Distribution/System.hs:89]: that comma after `used:` could probably go, too?

**Comment by Kleidukos** [comment]
@angerman to be absolutely honest I am all ears for a better way of expressing all the subtleties that we miss in the current nomenclature. First weird thing is that we conflate OS and Kernel, so Linux is a member, and GNU is an alias for Hurd, but Darwin is an alias for OSX, not IOS, and Android is its own platform. 

I understand that the distinctions are only the ones that we find relevant at a given time, but that looks like a huge footgun that is just waiting to happen.

**Comment by Bodigrim** [comment]
This change requires a new version of `cabal-format` and proper guarding in parser if `cabal-format` is lower.

**Comment by angerman** [comment]
> @angerman to be absolutely honest I am all ears for a better way of expressing all the subtleties that we miss in the current nomenclature. First weird thing is that we conflate OS and Kernel, so Linux is a member, and GNU is an alias for Hurd, but Darwin is an alias for OSX, not IOS, and Android is its own platform.
> 
> I understand that the distinctions are only the ones that we find relevant at a given time, but that looks like a huge footgun that is just waiting to happen.

I agree, this is indeed very unfortunate. We really want to distinguish kernel/userland most of the time. linux/gnu (glibc), linux/musl, linux/bionic (android), ...
For macOS, this is moving pretty much to the same though. The difference between macOS/iOS/tvOS/watchOS is shrinking by the day. It's all xnu/libSystem really. 

I don't know what the right solution here is. I'm just afraid to see `macos`, when people really mean one of apples platforms, not necessarily the one running on desktops/laptops. Currently drawin abstracts that fairly well. 🤷‍♂️

**Comment by ulysses4ever** [comment]
@Kleidukos you requested my review -- I'm grateful for the trust! But I think I should refrain. First, I have no experience with OS X and don't have any idea what osx v. darvin v. macos entails. Second, reading the thread, it seems to me like it's not as clear of a cut... In general, I prefer to burn bridges and right wrongs instead of cautiously maintaining all the legacy, but I'd much rather this PR was approved by someone at least remotely concerned with Mac.

**Comment by ulysses4ever** [comment]
Should this be pursued still?

**Comment by Kleidukos** [comment]
So, I guess my opinion would be that we have darwin and osx as its alias, not the opposite. But that's an uphill battle for breaking changes that I don't have the courage to lead. Perhaps a next time.

**Pull request closed without merging**